modules/tinymce/src/themes/silver/main/ts/ui/general/AlertBanner.ts

```diff
@@ -26,27 +28,26 @@ export const renderAlertBanner = (spec: AlertBannerWrapper, providersBackstage:
       {
         dom: {
           tag: 'div',
-          classes: [ 'tox-notification__icon' ]
+          classes: [ 'tox-notification__icon' ],
+          innerHtml: !spec.url ? icon : undefined
         },
-        components: [
+        components: spec.url ? [
           Button.sketch({
             dom: {
               tag: 'button',
               classes: [ 'tox-button', 'tox-button--naked', 'tox-button--icon' ],
-              innerHtml: Icons.get(spec.icon, providersBackstage.icons),
+              innerHtml: icon,
               attributes: {
                 title: providersBackstage.translate(spec.iconTooltip)
               }
             },
             // TODO: aria label this button!
-            action: (comp) => {
-              AlloyTriggers.emitWith(comp, formActionEvent, { name: 'alert-banner', value: spec.url });
-            },
+            action: (comp) => AlloyTriggers.emitWith(comp, formActionEvent, { name: 'alert-banner', value: spec.url }),
             buttonBehaviours: Behaviour.derive([
               Icons.addFocusableBehaviour()
             ])
           })
-        ]
+        ] : []
```

Comment:
Can components be undefined rather than an empty array?

Reply:
Fixed